### PR TITLE
Obsolete the legacy persistence service.

### PR DESCRIPTION
**Pull request by CyrusNajmabadi**
validating with F# @TIHan  and TypeScript @amcasey that these are not used by our direct partners.

### CODE REVIEW

File: src/Workspaces/Core/Portable/Workspace/Host/PersistentStorage/IPersistentStorageService.cs [hunk 1]
```diff
@@ -9,12 +9,15 @@
 namespace Microsoft.CodeAnalysis.Host
 {
     /// <summary>
-    /// This service allows you to persist information relative to solution, projects and documents.
+    /// Obsolete.  Roslyn no longer supports a mechanism to perform arbitrary persistence of data.  If such functionality
+    /// is needed, consumers are resonsible for providing it themselves with whatever semantics are needed.
```
Comment (JoeRobich) [inline src/Workspaces/Core/Portable/Workspace/Host/PersistentStorage/IPersistentStorageService.cs:13]:
```suggestion
    /// is needed, consumers are responsible for providing it themselves with whatever semantics are needed.
```

File: src/Workspaces/Core/Portable/Storage/LegacyPersistentStorageService.cs [hunk 2]
```diff
@@ -12,10 +12,10 @@
 namespace Microsoft.CodeAnalysis.Storage
 {
     /// <summary>
-    /// Implements public <see cref="IPersistentStorageService"/>.
-    /// Internally, it is preferred to work directly with <see cref="IChecksummedPersistentStorageService"/>,
-    /// which can be retrieved by <see cref="PersistentStorageExtensions.GetPersistentStorageService(HostWorkspaceServices, StorageDatabase)"/>.
+    /// Obsolete.  Roslyn no longer supports a mechanism to perform arbitrary persistence of data.  If such functionality
+    /// is needed, consumers are resonsible for providing it themselves with whatever semantics are needed.
```
Comment (JoeRobich) [inline src/Workspaces/Core/Portable/Storage/LegacyPersistentStorageService.cs:16]:
```suggestion
    /// is needed, consumers are responsible for providing it themselves with whatever semantics are needed.
```

**Review by tmat — APPROVED** [review]
:shipit:

**Comment by CyrusNajmabadi** [comment]
Have confirmed with Will that this is not used in F#.  Have checked the TS repo myself and have found no cases there.

**Review by amcasey — APPROVED** [review]
Thanks for checking!

**Pull request merged**